File: bestia-zoneserver/src/test/java/net/bestia/zoneserver/battle/ElementModifierTest.java
package net.bestia.zoneserver.battle;

import org.junit.Assert;
import org.junit.Test;

import net.bestia.model.domain.Element;

public class ElementModifierTest {

	@Test
	public void getModifier_differentTests() {

		Assert.assertEquals(100, ElementModifier.INSTANCE.getModifier(Element.EARTH, Element.EARTH));
		Assert.assertEquals(0, ElementModifier.INSTANCE.getModifier(Element.FIRE, Element.WATER_3));
		Assert.assertEquals(100, ElementModifier.INSTANCE.getModifier(Element.POISON, Element.NORMAL_2));
		Assert.assertEquals(0, ElementModifier.INSTANCE.getModifier(Element.GHOST, Element.NORMAL_2));
		Assert.assertEquals(-75, ElementModifier.INSTANCE.getModifier(Element.SHADOW, Element.UNDEAD_3));
	}

	@Test(expected = IllegalArgumentException.class)
	public void getModifier_nonLv1AttackElement_throws() {
		ElementModifier.INSTANCE.getModifier(Element.EARTH_3, Element.FIRE_2);
	}
}

File: bestia-zoneserver/src/test/kotlin/net/bestia/zoneserver/battle/ElementModifierTest.kt
package net.bestia.zoneserver.battle

import net.bestia.model.battle.Element
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test

class ElementModifierTest {

  @Test
  fun getModifier_differentTests() {
    assertEquals(100, ElementModifier.getModifier(Element.EARTH, Element.EARTH).toLong())
    assertEquals(0, ElementModifier.getModifier(Element.FIRE, Element.WATER_3).toLong())
    assertEquals(100, ElementModifier.getModifier(Element.POISON, Element.NORMAL_2).toLong())
    assertEquals(0, ElementModifier.getModifier(Element.GHOST, Element.NORMAL_2).toLong())
    assertEquals(-75, ElementModifier.getModifier(Element.SHADOW, Element.UNDEAD_3).toLong())
  }

  @Test
  fun getModifier_nonLv1AttackElement_throws() {
    assertThrows(IllegalArgumentException::class.java) {
      ElementModifier.getModifier(Element.EARTH_3, Element.FIRE_2)
    }
  }
}
